src/dkg_polynomial.erl

```diff
@@ -35,6 +63,14 @@ mul(PolyA, PolyB) ->
                         add(Acc, Temp)
                 end, [], PolyB).
 
+cmp(PolyA, PolyB) ->
+    %% check f(x) - g(x) = 0,
+    %% check that each element is the same the original
+    %% and the degree of polyA = polyB
+    dkg_polynomial:is_zero(dkg_polynomial:sub(PolyA, PolyB)) andalso
+    is_equal(PolyA, PolyB) andalso
+    degree(PolyA) == degree(PolyB).
```

Comment:
Zip in the previous step will fail if the 2 lengths are not the same length.

Reply:
I'd do the degree check first, and remove the zero check?

Reply:
Agreed on the degree check first. So the _formal_ definition of whether f(x) = g(x) states that if f(x) - g(x) = h(x), then h(x) = 0 for all values of x. Hence the zero checks for both poly and bipoly.

Reply:
Why is that better than just straight element comparison, though? I can see that being useful in a mathematical way, but we don't subtract a number from itself to check if it's the same in programming?

Reply:
Fair enough, I was just being pedantic with the mathematical definition.